Sources/PrimerSDK/Classes/Extensions & Utilities/PrimerButton.swift

```diff
@@ -13,59 +13,183 @@ import UIKit
+        didSet {
+            if let image = imageLogo {
+                setImage(image, for: .normal)
+            }
+        }
     }
     
-    func showSpinner(_ flag: Bool, color: UIColor = .white) {
-        DispatchQueue.main.async {
-            if self.titleCopy == nil {
-                self.titleCopy = self.titleLabel?.text
+    internal var title: String? {
+        didSet {
+            setTitle(title, for: .normal)
+        }
+    }
+    
+    public var onPressed: Action?
+    
+    override open var isHighlighted: Bool {
+        didSet {
+            if !oldValue {
+                backgroundNormalStateColor = backgroundColor ?? .clear
             }
-            
-            self.isUserInteractionEnabled = !flag
-            
-            if self.spinner == nil {
-                self.spinner = UIActivityIndicatorView()
-                self.addSubview(self.spinner)
-                self.spinner.translatesAutoresizingMaskIntoConstraints = false
-                self.spinner.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
-                self.spinner.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
-                self.spinner.widthAnchor.constraint(equalToConstant: 20).isActive = true
-                self.spinner.heightAnchor.constraint(equalToConstant: 20).isActive = true
+            if let backgroundHighlightedColor = backgroundHighlightedStateColor {
+                backgroundColor = isHighlighted ? backgroundHighlightedColor : backgroundHighlightedStateColor
             }
-            
-            self.spinner.color = color
-            
-            flag ? self.spinner.startAnimating() : self.spinner.stopAnimating()
-            flag ? self.setTitle(nil, for: .normal) : self.setTitle(self.titleCopy, for: .normal)
-            self.spinner.isHidden = !flag
         }
     }
+    
+    //MARK: - Button States for Activity Indicator
+    
+    internal struct ActivityIndicatorButtonState {
+        var state: UIControl.State
+        var title: String?
+        var image: UIImage?
+    }
 
-    func pin(
-        to view: UIView,
-        leading: CGFloat = 0,
-        top: CGFloat = 0,
-        trailing: CGFloat = 0,
-        bottom: CGFloat = 0
-    ) {
-        topAnchor.constraint(equalTo: view.topAnchor, constant: top).isActive = true
-        bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: bottom).isActive = true
-        leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: leading).isActive = true
-        trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: trailing).isActive = true
+    private(set) var activityIndicatorButtonStates: [ActivityIndicatorButtonState] = []
+    
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.color = theme?.colorStates.color(for: .selected)
```

Comment:
The activity indicator color is light grey now (it was white before). Not sure if it makes sense to get the color for the selected state, or if it would be better to get the text color. I know that the previous behavior was just white, so can you provide the `white` color on the merchant's app example and check if it works?

Reply:
Good point on that. The reason why I went after the selected state is that I thought from the product perspective.
The point in which we animate the button is when we press it.
So I thought that would be good for consistency to keep the activity indicator color following **the selected state**.
In case we don't have a theme though, this won't work - so I'll do the following:

`activityIndicator.color = theme?.colorStates.color(for: .selected) ?? .white`

what do you think? @EvansPie